frame/name-service/src/lib.rs

```diff
@@ -0,0 +1,487 @@
+		NameRenewed { name_hash: NameHash, expires: T::BlockNumber },
+		/// An address has been set for a name hash to resolve to.
+		AddressSet { name_hash: NameHash, address: T::AccountId },
+		/// An address was deregistered.
+		AddressDeregistered { name_hash: NameHash },
+	}
+
+	// Your Pallet's error messages.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// This commitment hash already exists in storage.
+		AlreadyCommitted,
+		/// It has not passed the minimum waiting period to reveal a commitement.
+		TooEarlyToReveal,
+		/// The commitment cannot yet be removed. Has not expired.
+		CommitmentNotExpired,
+		/// This commitment does not exist.
+		CommitmentNotFound,
+		/// This registration does not exist.
+		RegistrationNotFound,
+		/// The sender is not the registration owner.
+		NotRegistrationOwner,
+		/// This registration has expired.
+		RegistrationExpired,
+		/// This registration has not yet expired.
+		RegistrationNotExpired,
+	}
+
+	// Your Pallet's callable functions.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		// TODO: Should we allow registration on behalf of?
+		#[pallet::weight(0)]
+		pub fn commit(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			commitment_hash: CommitmentHash,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+
+			ensure!(!Commitments::<T>::contains_key(commitment_hash), Error::<T>::AlreadyCommitted);
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+			let deposit = T::CommitmentDeposit::get();
+
+			T::Currency::reserve(&sender, deposit)?;
+
+			let commitment = Commitment { who: who.clone(), when: block_number, deposit };
+
+			Commitments::<T>::insert(commitment_hash, commitment);
+			Self::deposit_event(Event::<T>::Committed { sender, who, hash: commitment_hash });
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn reveal(
+			origin: OriginFor<T>,
+			name: Vec<u8>,
+			secret: u64,
+			periods: u32,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			let commitment_hash = sp_io::hashing::blake2_256(&(&name, secret).encode());
+
+			let commitment =
+				Commitments::<T>::get(commitment_hash).ok_or(Error::<T>::CommitmentNotFound)?;
+
+			let name_hash = sp_io::hashing::blake2_256(&name);
+			let block_number = frame_system::Pallet::<T>::block_number();
+
+			ensure!(
+				block_number > commitment.when.saturating_add(T::MinimumCommitmentPeriod::get()),
+				Error::<T>::TooEarlyToReveal
+			);
+
+			if Self::is_available(name_hash, block_number) {
+				let fee = Self::registration_fee(name.clone(), periods);
+
+				let imbalance = T::Currency::withdraw(
+					&sender,
+					fee,
+					WithdrawReasons::FEE,
+					ExistenceRequirement::KeepAlive,
+				)?;
+
+				T::RegistrationFeeHandler::on_unbalanced(imbalance);
+
+				let deposit = T::NameDeposit::get();
+				Self::do_register(name_hash, commitment.who.clone(), deposit, periods)?;
+			}
+
+			T::Currency::unreserve(&commitment.who, commitment.deposit);
```

Comment:
Currently the commitment deposit is constant, so you dont need to save it, right?

Reply:
you MUST save it, else if we ever update it, then we will have some inconsistent state. Users with an old deposit amount should have on chain reflect that have some different deposit amount.